### ISSUE TITLE
Enabling caching looks not working (keeps disable after click switch)

### ISSUE THREAD

**Issue by mglaredo** [metabase/metabase#15828]
**Describe the bug**
Trying to use CACHING as explained here, the switch keeps off and parameters doesn't appear (e.g. _Minimum query duration_) to customize the question cache.

**Logs**
No javascript or log info when clicking to try toggling Caching on

**To Reproduce**
1. Go to Setup 
2. Click on left menu: caching section
3. Click switch for enabling. 
4. Saved green message on right corner appears but no parameters and keep disabled.

**Expected behavior**
To appear caching parameters and to be able of keep switch on

**Screenshots**
![image](https://user-images.githubusercontent.com/1532294/116242749-85e7dd00-a766-11eb-9cdf-88021d0a4826.png)

**Severity**
I am looking for improving Metabase dashboard performance, so caching is a good aspect to solve on my installation.

**Additional context**
Same for current version (v0.39.0.1) and previous one (v0.38.1)

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:87.0) Gecko/20100101 Firefox/87.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_272-b10",
    "java.vendor": "Red Hat, Inc.",
    "java.vendor.url": "https://www.redhat.com/",
    "java.version": "1.8.0_272",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.272-b10",
    "os.name": "Linux",
    "os.version": "4.18.0-193.6.3.el8_2.x86_64",
    "user.language": "es",
    "user.timezone": "Europe/Madrid"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MariaDB",
        "version": "10.3.17-MariaDB"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.6.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.39.0.1",
      "date": "2021-04-20",
      "branch": "release-x.39.x",
      "hash": "47bb5f2"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Comment by flamber** [comment]
Hi @mglaredo 
Are you running multiple instances of Metabase with a reverse-proxy without sticky connections?
Or have you defined the setting via environment variables?

**Comment by mglaredo** [comment]
Hi @flamber. No reverse-proxy on our installation and yes, we stablish our MB variables as environment exports.

**Comment by flamber** [comment]
@mglaredo If you have define the cache via environment variables like https://www.metabase.com/docs/latest/operations-guide/environment-variables.html#mb_enable_query_caching, then you cannot change that in the UI.
There's a bug with all toggles defined via environment variables, where it shows as disabled in the UI even though it's enabled.
Likely a duplicate of #13166 / #14790

**Comment by mglaredo** [comment]
Sorry, I meant that we setup several variables "MB_*" before starting JAR. But, caching is not one of them. Therefore, we're going to UI setup inside metabase looking for enabling this aspect. I am going to try setting as enviroment variable MB_ENABLE_QUERY_CACHING and I'll give you feedback then.
Thanks

**Comment by flamber** [comment]
@mglaredo You should not be using environment variables for that.

I'm 99% sure that a reverse-proxy is in play - my first thought was a load-balancer, since that would be the typical problem, but it could also be some aggressive cache/CDN.

But please post the data from request `PUT /api/setting/enable-query-caching` - use your browser developer Network-tab to see.
Also include the logs too for when you're trying to change things - Admin > Troubleshooting > Logs.

**Comment by mglaredo** [comment]
Hi @flamber, I thought that only set basic MB_* enviroment variables before JAR launch, but certainly export MB_ENABLE_QUERY_CACHING=true was there. I comment out the line and restart JAR. Now, the UI for caching is okay. Thanks!

### These are the post data
#### HEADERS
PUT /api/setting/enable-query-caching HTTP/1.1
Host: cauceprud2.ugr.es:8443
Connection: keep-alive
Content-Length: 454
Accept: application/json
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.96 Safari/537.36
Content-Type: application/json
Origin: https://cauceprud2.ugr.es:8443
Sec-Fetch-Site: same-origin
Sec-Fetch-Mode: cors
Sec-Fetch-Dest: empty
Referer: https://cauceprud2.ugr.es:8443/admin/settings/caching
Accept-Encoding: gzip, deflate, br
Accept-Language: es-ES,es;q=0.9
Cookie: _ga=GA1.2.597209189.1619003937; metabase.DEVICE=5620cff9-5920-45d2-b48a-b47a604ed515; metabase.SESSION=59f5ae10-2f8c-4d41-8a8a-088503e18e85; _gid=GA1.2.334134507.1619593615
#### PLAYLOAD
{"placeholder":"Utilizando el valor de la variable de entorno $MB_ENABLE_QUERY_CACHING","value":true,"is_env_setting":true,"env_name":"MB_ENABLE_QUERY_CACHING","description":"Al habilitar el almacenamiento en caché se guardan los resultados de las consultas que tardan mucho tiempo en ejecutarse.","default":"Utilizando el valor de la variable de entorno $MB_ENABLE_QUERY_CACHING","originalValue":null,"display_name":"Habilitar caché","type":"boolean"}

### SERVER LOG
[cbfaf425-100c-471b-868d-f0ff0008d78a] 2021-04-28T10:24:29+02:00 DEBUG metabase.server.middleware.log PUT /api/setting/enable-query-caching 204 50,5 ms (4 llamadas a la BBDD) Conexiones BBDD: 1/15 Hilos Jetty: 5/50 (2 esperando, 0 en cola) (124 hilos totales activos) Consultas en ejecución: 0 (0 en cola)
[cbfaf425-100c-471b-868d-f0ff0008d78a] 2021-04-28T10:24:29+02:00 DEBUG metabase.server.middleware.log GET /api/setting 200 3,2 ms (0 llamadas a la BBDD) Conexiones BBDD: 1/15 Hilos Jetty: 5/50 (2 esperando, 0 en cola) (124 hilos totales activos) Consultas en ejecución: 0 (0 en cola)
[cbfaf425-100c-471b-868d-f0ff0008d78a] 2021-04-28T10:24:29+02:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 13,1 ms (2 llamadas a la BBDD) Conexiones BBDD: 1/15 Hilos Jetty: 7/50 (1 esperando, 0 en cola) (125 hilos totales activos) Consultas en ejecución: 0 (0 en cola)
[cbfaf425-100c-471b-868d-f0ff0008d78a] 2021-04-28T10:24:35+02:00 DEBUG metabase.server.middleware.log GET /api/util/bug_report_details 200 7,4 ms (1 llamadas a la BBDD) Conexiones BBDD: 2/15 Hilos Jetty: 5/50 (2 esperando, 0 en cola) (127 hilos totales activos) Consultas en ejecución: 0 (0 en cola)
